File: Truncate-Sentence.py
class Solution(object):
    def truncateSentence(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        ans = ""
        n = len(s)
        for i in range(n):
            if(s[i] == ' '):
                k = k - 1
            if (k == 0):
                break
            else:
                ans = ans + s[i]
        return ans

File: Truncate-Sentence.java
class Solution {
    public String truncateSentence(String s, int k) {
        StringBuilder ans = new StringBuilder();
        int n = s.length();
        for (int i = 0; i < n; i++) {
            if (s.charAt(i) == ' ') {
                k--;
            }
            if (k == 0) {
                break;
            } else {
                ans.append(s.charAt(i));
            }
        }
        return ans.toString();
    }
}